DigitalLearningSolutions.Web/ViewModels/TrackingSystem/CourseSetup/SearchableCourseStatisticsViewModel.cs

```diff
@@ -1,28 +1,39 @@
 ﻿namespace DigitalLearningSolutions.Web.ViewModels.TrackingSystem.CourseSetup
 {
+    using System;
     using DigitalLearningSolutions.Data.Models.Courses;
+    using DigitalLearningSolutions.Web.Helpers;
+    using DigitalLearningSolutions.Web.ViewModels.Common.SearchablePage;
 
-    public class SearchableCourseStatisticsViewModel
+    public class SearchableCourseStatisticsViewModel : BaseFilterableViewModel
     {
         public SearchableCourseStatisticsViewModel(CourseStatistics courseStatistics)
         {
             CustomisationId = courseStatistics.CustomisationId;
-            CentreId = courseStatistics.CentreId;
-            Active = courseStatistics.Active;
             DelegateCount = courseStatistics.DelegateCount;
-            CompletedCount = courseStatistics.CompletedCount;
             InProgressCount = courseStatistics.InProgressCount;
-            PassRate = courseStatistics.PassRate;
             CourseName = courseStatistics.CourseName;
+            CategoryName = courseStatistics.CategoryName;
+            LearningMinutes = courseStatistics.LearningMinutes;
+            Tags = FilterableTagHelper.GetCurrentTagsForCourseStatistics(courseStatistics);
         }
 
         public int CustomisationId { get; set; }
-        public int CentreId { get; set; }
-        public bool Active { get; set; }
         public int DelegateCount { get; set; }
-        public int CompletedCount { get; set; }
         public int InProgressCount { get; set; }
         public string CourseName { get; set; }
-        public double PassRate { get; set; }
+        public string CategoryName { get; set; }
+        public string LearningMinutes { get; set; }
+
+        public string EmailHref => GenerateEmailHref();
+
+        private string GenerateEmailHref()
+        {
+            var launchCourseLink =
+                $"{ConfigHelper.GetAppConfig()[ConfigHelper.AppRootPathName]}/LearningMenu/{CustomisationId}";
+            var subject = Uri.EscapeDataString($"Digital Learning Solutions Course Link - {CourseName}");
+            var content = Uri.EscapeDataString($"To start your {CourseName} course, go to {launchCourseLink}");
+            return $"mailto:?subject={subject}&body={content}";

```

Comment:
Is there a reason we don't do UrlEncode on the whole url instead here (not sure how it interacts with mailto links though - just interested 🤔)?

Reply:
The reason the UrlEncode is done in the first place is because mailto requires a few specific characters to function correctly so it knows when to split things (e.g. "&body=" specifies the email body). If you don't encode your subject/content and it contains one of the reserved characters, it can break things.

For instance, if a course name was something like "Excel & Word Office Training" then the email subject would appear as "Digital Learning Solutions Course Link - Excel " as it interprets "&..." as a new header. 

You also can't encode the entire href because it will then encode those specific characters used to specify the different sections. It no longer becomes recognisable as a mailto, and instead is interpreted as a site URL. 

So you have to encode the actual stuff that goes in the email and nothing else basically.